### ISSUE TITLE
Normalization of Strings containing - results in wrong query

### ISSUE THREAD

**Issue by ridono** [querydsl/querydsl#2017]
I have an issue when using setUseLiterals(true) in combination with a query that contains the parameter 'AR-1-5-3'. This results in a wrong query, where the parameter is normalized to 'A-6-3'. The Normalizer detects a mathematical expression in the substring and evaluates it.

`query.from(MY_TABLE).where(MY_TABLE.name.eq("AR-1-5-3"));`

Results in the sql:
`from my_table my_table 
where my_table .name = 'AR-6-3'`

instead of:
`from my_table my_table 
where my_table .name = 'AR-1-5-3'`

I found this behaviour up to version 3.7.4 using QueryDSL for SQL.

**Comment by Shredder121** [comment]
Thanks for the report.
We're already trying to figure out how to fix this issue (operations too far away from the `'`).
Related pr #1974.

**Comment by jwgmeligmeyling** [comment]
I am not sure if this issue is still present. The below test passes:

```java
@Test
public void useLiterals() {
    SQLSerializer serializer = new SQLSerializer(Configuration.DEFAULT);
    serializer.setUseLiterals(true);

    SimplePath<String> name = Expressions.path(String.class, Expressions.path(Object.class, "myTable"), "name");
    BooleanExpression predicate = name.eq("AR-1-5-3");

    serializer.handle(predicate);
    assertEquals("myTable.name = 'AR-1-5-3'", serializer.toString());
}
```

What type does your query have?

**Comment by jwgmeligmeyling** [comment]
My test reproduces the issue on 3.7.4. That means the issue is fixed somewhere between 3.7.4 and the current 4.3.1.

I encourage anyone still suffering from this issue (after four years...) to upgrade to a newer version of QueryDSL.
